### PR TITLE
17.0 mig base substate

### PR DESCRIPTION
**Pull request by Wodran14**

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@OCA/core-maintainers

**Review by ferran-S73 — APPROVED** [review]
Functional and code review

**Comment by StefanRijnhart** [comment]
Is there anything to be done about this warning?

```
2024-05-16T11:00:39.7512810Z 2024-05-16 11:00:39,750 249 [1;33m[1;49mWARNING[0m odoo odoo_test_helper.fake_model_loader: Wrong Import in module base_substate, the class <class 'odoo.addons.base_substate.tests.sale_test.SaleTest'> have been already imported.
2024-05-16T11:00:39.7514942Z Please take a look to the README on how to import test class 
2024-05-16T11:00:39.7516883Z 2024-05-16 11:00:39,750 249 [1;33m[1;49mWARNING[0m odoo odoo_test_helper.fake_model_loader: Wrong Import in module base_substate, the class <class 'odoo.addons.base_substate.tests.sale_test.LineTest'> have been already imported.
2024-05-16T11:00:39.7518675Z Please take a look to the README on how to import test class 
```

**Review by CRogos — COMMENTED** [review]
Are you still working on this?

Could you cleanup the commit history according to the migration guide?
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

image

File: base_substate/models/base_substate_mixin.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    def _track_template(self, tracking):
+        res = super()._track_template(tracking)
+        first_rec = self[0]
+        changes, tracking_value_ids = tracking[first_rec.id]
+        if "substate_id" in changes and first_rec.substate_id.mail_template_id:
+            res["substate_id"] = (
+                first_rec.substate_id.mail_template_id,
+                {
+                    "auto_delete_message": True,
```
Comment (ellbristow) [inline base_substate/models/base_substate_mixin.py:39]: auto_delete_message has been renamed to auto_delete in 17.0

**Review by ellbristow — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
Superseded by #940

**Pull request closed without merging**